common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Config
+    # Filters versions that should not be considered for dependency updates
+    class IgnoreCondition
+      UPDATE_TYPES = %i(
+        ignore_major_versions
+        ignore_minor_versions
+        ignore_patch_versions
+      ).freeze
+
+      ALL_VERSIONS = ">= 0"
+
+      attr_reader :dependency_name, :versions, :update_types
+      def initialize(dependency_name:, versions: nil, update_types: nil)
+        @dependency_name = dependency_name
+        @versions = versions || []
+        @update_types = update_types || []
+      end
+
+      def ignored_versions(dependency)
+        return [ALL_VERSIONS] if @versions.empty? && @update_types.empty?
+
+        versions_by_type(dependency) + @versions
+      end
+
+      private
+
+      def effective_update_type
```

Comment:
Question: should `update_type` be a single value?

I assumed we don't need to care about every range, and just need to focus on the most restrictive (e.g. major > minor > patch).
It might make sense for the parser to select the "effective" `update_type`, and let `IgnoreCondition` deal with a single value. 🤷

Reply:
I think we do need to care about each range? For the case of ignoring `version-update:semver-major` we were assuming the user still wants to get the minor/patch releases unless they also opt out of those types specifically. To be consistent in how it works the user could also specify something like `[version-update:semver-major, version-update:semer-patch]` to only get minor releases but not major/patch.

Sorry, I should have spotted this earlier!